server/channels/store/sqlstore/post_store.go

```diff
@@ -1043,14 +1056,30 @@ func (s *SqlPostStore) PermanentDeleteByChannel(channelId string) (err error) {
 	}
 	defer finalizeTransactionX(transaction, &err)
 
-	results := []postIds{}
-	err = transaction.Select(&results, "SELECT Id, RootId, UserId FROM Posts WHERE ChannelId = ?", channelId)
+	ids := []string{}
+	err = transaction.Select(&ids, "SELECT Id FROM Posts WHERE ChannelId = ?", channelId)
 	if err != nil {
 		return errors.Wrapf(err, "failed to fetch Posts with channelId=%s", channelId)
 	}
 
-	for _, ids := range results {
-		if err = s.permanentDeleteThreads(transaction, ids.Id); err != nil {
+	batchIds := [][]string{}
+	batchSize := 1000
+	resultsSize := len(ids)
+
+	for i := 0; i < resultsSize; i += batchSize {
+		end := i + batchSize
+
+		if end > resultsSize {
+			end = resultsSize
+		}
+
+		batchIds = append(batchIds, ids[i:end])
+	}
+	for _, batch := range batchIds {
+		if err = s.permanentDeleteThreads(transaction, batch); err != nil {
```

Comment:
Should we have a short sleep between batches to avoid pegging the CPU?

Reply:
Yeah I can do that

Reply:
I wasn't exactly sure what a good sleep time might be, 100 ms felt like a lot for this endpoint because it does a lot of different deletes and is already pretty slow overall. I went with 10ms but do we have any recommendations for what an optimal time is?

Reply:
10ms is fine IMO.